core/app/models/spree/order.rb

```diff
@@ -782,11 +782,12 @@ def process_payments_before_complete
     #   }
     #
     def update_params_payment_source
-      if @updating_params[:order] && (@updating_params[:order][:payments_attributes] || @updating_params[:order][:existing_card])
+      if @updating_params&[:order] && (@updating_params&[:order]&[:payments_attributes] || @updating_params&[:order]&[:existing_card])
```

Comment:
Better figure out how to set some dummy data to this instance variable `@updating_params` in the test rather than changing the method. Given that it hasn't been tested so far, any change can break it really.